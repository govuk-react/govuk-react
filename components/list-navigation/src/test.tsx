import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import ListNavigation from '.';

const example = 'example';
const wrapper = <ListNavigation>{example}</ListNavigation>;
const wrapperMultiple = (
  <ListNavigation>
    {example}
    {example}
  </ListNavigation>
);

describe('ListNavigation', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListNavigation>{example}</ListNavigation>, div);
  });

  it('should render an unordered list', () => {
    const output = mount(wrapper);
    expect(output.find('ul')).toHaveLength(1);
    expect(mount(wrapperMultiple).find('ul li')).toHaveLength(2);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(<ListNavigation listStyleType="square">Example</ListNavigation>)).toMatchSnapshot('wrapper mount');
  });
});
