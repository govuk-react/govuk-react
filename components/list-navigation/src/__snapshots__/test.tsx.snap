// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListNavigation matches wrapper snapshot: wrapper mount 1`] = `
.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 5px;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  list-style-type: square;
  padding-left: 20px;
}

.c0 .c0 {
  margin-top: 10px;
}

.c1 > .c2 {
  margin-bottom: 5px;
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

<ListNavigation
  listStyleType="square"
>
  <UnorderedList
    listStyleType="square"
  >
    <OrderedList
      as="ul"
      listStyleType="square"
    >
      <ul
        className="c0 c1"
      >
        <ListItem>
          <li
            className="c2 c3"
          >
            Example
          </li>
        </ListItem>
      </ul>
    </OrderedList>
  </UnorderedList>
</ListNavigation>
`;
